apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ErrorCode.cs

```diff
@@ -17,70 +17,100 @@
 namespace Google.Cloud.Spanner.Data
 {
     /// <summary>
+    /// A detailed code supplied via <see cref="SpannerException.ErrorCode"/> when an error
+    /// occurs.
     /// </summary>
     public enum ErrorCode
     {
         /// <summary>
+        /// The operation was cancelled.
         /// </summary>
         Cancelled = StatusCode.Cancelled,
 
         /// <summary>
+        /// Unknown error. This may happen if an internal error occured or
+        /// not enough information was obtained to produce a useful message.
         /// </summary>
         Unknown = StatusCode.Unknown,
 
         /// <summary>
+        /// A bad SQL statement or other invalid input was sent to Spanner.
         /// </summary>
         InvalidArgument = StatusCode.InvalidArgument,
 
         /// <summary>
+        /// A timeout has occurred. A <see cref="SpannerTransaction"/> should be
+        /// retried if this error is encountered.
         /// </summary>
         DeadlineExceeded = StatusCode.DeadlineExceeded,
 
         /// <summary>
+        /// The given resource (Spanner Instance, Database, Table) was not
+        /// found.
         /// </summary>
         NotFound = StatusCode.NotFound,
 
         /// <summary>
+        /// The Spanner object could be not created because it already exists.
         /// </summary>
         AlreadyExists = StatusCode.AlreadyExists,
 
         /// <summary>
+        /// The supplied credential from <see cref="SpannerConnectionStringBuilder.Credential"/>
+        /// or from default application credentials does not have sufficient permission for the request.
         /// </summary>
         PermissionDenied = StatusCode.PermissionDenied,
 
         /// <summary>
+        /// There is no supplied credential either through default application credentials or
+        /// directly through <see cref="SpannerConnectionStringBuilder.Credential"/>
         /// </summary>
         Unauthenticated = StatusCode.Unauthenticated,
 
         /// <summary>
+        /// <see cref="SpannerOptions.MaximumActiveSessions"/> has been reached.
+        /// Therefore a new session with Spanner could not be created.
```

Comment:
Right now, its the only possible way to get this error.  server side resources, when exhausted will give a 5XX error which is retry-able.
We could make this vague to cover ourselves against future changes -- but we could just as easily touch up the description when we do have another path.